NTupleMaker/bin/SynchNTupleProducer_2017.cpp

```diff
@@ -68,16 +68,30 @@
 #define pionMass 		   0.1396
 
 
-
 void FillMuTau(const AC1B * analysisTree, Synch17Tree *otree, int leptonIndex, float dRiso);
 void FillETau(const AC1B * analysisTree, Synch17Tree *otree, int leptonIndex, float dRiso);
 void FillTau(const AC1B * analysisTree, Synch17Tree *otree, int tauIndex);
 void FillTau_leading(const AC1B * analysisTree, Synch17Tree *otree, int tauIndex);
 void initializeCPvar(Synch17Tree *otree);
+void SaveRECOVertices(const AC1B * analysisTree,Synch17Tree *otree, const bool isData);
 void initializeGenTree(Synch17GenTree *gentree);
 void FillGenTree(const AC1B * analysisTree, Synch17GenTree *gentree, TString ch);
 //void fillTTbarUncWeights(const AC1B * analysisTree, Synch17Tree *otree, bool isData, bool includeTTbarUncWeights);
 
+
+/*Notifications Merijn
+-to my experience currently macro only works for mu-tau (unless udpated in the meantime)
+-added a histogram and leafs in the otree to keep track of the pdg codes of the hadronically decaying tau, and ctr to keep track how often 2nd tau does NOT decay hadronically
+-we call an improved CP function, in which a number of bugs are solved. It takes the decay mode also as argument, to construct certain kinematic variables
+-Merijn suspects that second reco tau in mu-tau is currently not necessarily hadronically decaying, since RECO decay products substatial fraction 
+-see functionsCP.h for updates on the CP calculation
+*/
+
+
+//Merijn added a histogram to spot the pdg codes of the decaying hadronic tau
+TH1F* ConstitsPDG=new TH1F("ConstitsPDG","ConstitsPDG",500,-250,250);
+int nonpionphotonctr=0;
+
```

Comment:
I would highly recommend moving these instructions in the `main()` maybe around L 505 where the `inputEventsH` and `nWeightedEventsH` histograms are defined.

Reply:
Done